compiler/rustc_typeck/src/check/wfcheck.rs

```diff
@@ -163,6 +163,23 @@ pub fn check_item_well_formed(tcx: TyCtxt<'_>, def_id: LocalDefId) {
                 }
             }
         }
+        hir::ItemKind::TyAlias(ref ty, hir::Generics { params: [], .. }) => {
+            // We explicitely only check cases with *no* type parameters to avoid regressions on
+            // existing incorrectly accepted code like `type T<X> = <X as Trait>::Foo;`.
+
+            // We don't use `check_item_type` to *not* emit `!Sized` errors.
+            let ty_span = ty.span;
+            for_id(tcx, item.hir_id(), ty_span).with_fcx(|fcx, tcx| {
```

Comment:
After https://github.com/rust-lang/rust/pull/81339 we'll have the param env reveal mode `Selection` which allows us to suppress const eval errors entirely. It may be of interest to use that here to not break existing code (as shown in various tests). Though we should probably extend the `const_err` lint to also cover constants in type aliases independently of this.